lib/eq3interface.js

```diff
@@ -11,60 +11,301 @@ const status = {
+      out[5] = date.getMinutes();
+      out[6] = date.getSeconds();
+      return out;
     },
+
+    getDay: (day) => {
+      return new Buffer('200' + day, 'hex');
+    },
+
+    // set schedule for a day
+    // day is { day: <daynum, 0=sat>, segments:[7 x {temp:<temp>, endtime:{ hour:<hour>, min:<min>}}, ...]}
+    setDay: (day) => {
+      var out = new Buffer(16);
+      out[0] = 0x10;
+      out[1] = day.day;
+
+      // zero all first
+      for (var i = 0; i < 7; i++) {
+        out[(i * 2) + 2] = 0;
+        out[(i * 2) + 3] = 0;
+      }
+
+      for (var i = 0; i < 7; i++) {
+        out[(i * 2) + 2] = 0;
+        out[(i * 2) + 3] = 0;
+
+        if (day.segments[i].temp &&
+          day.segments[i].endtime &&
+          (day.segments[i].endtime.hour !== undefined) &&
+          (day.segments[i].endtime.min !== undefined)) {
+          out[(i * 2) + 2] = (day.segments[i].temp * 2) >> 0;
+          out[(i * 2) + 3] = (((day.segments[i].endtime.hour * 60) + day.segments[i].endtime.min) / 10) >> 0;
+        } else {
+          break; // stop at first non-temp
+        }
+      }
+      return out;
+    }
+
   },
-  parseInfo: function(info) {
-    const statusMask = info[2]
-    const valvePosition = info[3]
-    const targetTemperature = info[5] / 2
 
+  ////////////////////////////////////////////////
+  // start of parse functions.
+  // these parse the response data - send as notify
+  //
+  // don't know what info[0] = 0 could mean, or remember if I've ever seen it 
+  parseInfo_00: function(info) {
     return {
+      unknown: true,
+      raw: info,
+    };
+  },
+
+  // sysinfo
+  parseSysInfo: function(info) {
+    return {
+      sysinfo: {
+        ver: info[1],
+        type: info[2],
+      },
+      raw: info,
+    };
+  },
+
+  // for 02x1 responses
+  parseStatus: function(info) {
+    const statusMask = info[2];
+    const valvePosition = info[3];
+    const targetTemperature = info[5] / 2;
+
+    var ecoendtime = undefined;
+    if (((statusMask & status.holiday) === status.holiday) && (info.length >= 10)) {
+      // parse extra bytes
+      var ecotime = {
+        day: info[6],
+        year: info[7] + 2000,
+        hour: (info[8] / 2) >> 0,
+        min: (info[8] & 1) ? 30 : 0,
+        month: info[9],
+      };
+      ecoendtime = new Date(ecotime.year, ecotime.month - 1, ecotime.day, ecotime.hour, ecotime.min, 0, 0);
+    }
+
+    return {
+      raw: info,
       status: {
         manual: (statusMask & status.manual) === status.manual,
         holiday: (statusMask & status.holiday) === status.holiday,
         boost: (statusMask & status.boost) === status.boost,
+        lock: (statusMask & status.lock) === status.lock,
         dst: (statusMask & status.dst) === status.dst,
         openWindow: (statusMask & status.openWindow) === status.openWindow,
         lowBattery: (statusMask & status.lowBattery) === status.lowBattery,
+        valvePosition,
+        targetTemperature,
```

Comment:
ahh..  just so as not to break someone else who was using the output; yet they were 'in the wrong place' in my thinking, so I wanted to move them gracefully.

Reply:
I'll see your point. Nothing wrong with changing things. Probably it was not wise to name the group `status` since everything here can be considered status 🤨
I see no harm if it stays how it is.